src/Libraries/LibAllowList.sol

```diff
@@ -1,66 +1,86 @@
 // SPDX-License-Identifier: LGPL-3.0-only
-/// @custom:version 1.0.0
 pragma solidity ^0.8.17;
 
 import { InvalidContract } from "../Errors/GenericErrors.sol";
+import { LibAsset } from "./LibAsset.sol";
 
-/// @title Lib Allow List
+/// @title LibAllowList
 /// @author LI.FI (https://li.fi)
-/// @notice Library for managing and accessing the conract address allow list
+/// @notice Library for managing and accessing the contract address allow list
+/// @custom:version 1.0.1
 library LibAllowList {
     /// Storage ///
     bytes32 internal constant NAMESPACE =
         keccak256("com.lifi.library.allow.list");
 
     struct AllowListStorage {
-        mapping(address => bool) allowlist;
+        mapping(address => bool) contractAllowList;
         mapping(bytes4 => bool) selectorAllowList;
         address[] contracts;
+        mapping(address => uint256) contractToIndex;
+        mapping(bytes4 => uint256) selectorToIndex;
+        bytes4[] selectors;
+        bool migrated;
     }
 
     /// @dev Adds a contract address to the allow list
     /// @param _contract the contract address to add
     function addAllowedContract(address _contract) internal {
-        _checkAddress(_contract);
+        // validate contract address is not zero address
+        if (_contract == address(0)) revert InvalidContract();
```

Comment:
nitpick: I would vouch to use InvalidCalldata() here. 
InvalidContract() sounds misleading imho, but that's just a personal opinion.  No need to fix if you dont want to

Reply:
good point! updated